ncr/src/main/kotlin/com/hartwig/actin/personalization/ncr/interpretation/filter/TreatmentRecordFilter.kt

```diff
@@ -0,0 +1,166 @@
+package com.hartwig.actin.personalization.ncr.interpretation.filter
+
+import com.hartwig.actin.personalization.ncr.datamodel.NcrRecord
+import com.hartwig.actin.personalization.ncr.datamodel.NcrTreatment
+
+class TreatmentRecordFilter(override val logFilteredRecords: Boolean) : RecordFilter {
+
+    internal fun hasAtLeastOneTreatment(treatment: NcrTreatment): Boolean {
+        return listOf(
+            treatment.primarySurgery.chir,
+            treatment.primaryRadiotherapy.rt,
+            treatment.systemicTreatment.chemo,
+            treatment.hipec.hipec
+        ).any { it.notZeroNorNull() } ||
+                listOf(
+                    treatment.metastaticSurgery.metaChirCode1,
+                    treatment.metastaticSurgery.metaChirCode2,
+                    treatment.metastaticSurgery.metaChirCode3,
+                    treatment.metastaticRadiotherapy.metaRtCode1,
+                    treatment.metastaticRadiotherapy.metaRtCode2,
+                    treatment.metastaticRadiotherapy.metaRtCode3,
+                    treatment.metastaticRadiotherapy.metaRtCode4,
+                ).any { it != null }
+    }
+
+    internal fun hasConsistentReasonRefrainingTreatment(tumorRecords: List<NcrRecord>): Boolean {
+        val hasConsistentReasonRefrainingTreatment = tumorRecords.map { it.treatment }.all { treatment ->
+            when (hasAtLeastOneTreatment(treatment)) {
+                true -> treatment.geenTherReden == null
+                else -> true
+            }
```

Comment:
We should assert that `geenTherReden != null` if there's no treatment. I would use `if` instead of `when` here, or:
```suggestion
            hasAtLeastOneTreatment(treatment) == (treatment.geenTherReden == null)
```